app/src/behaviors/behavior_macro.c

```diff
@@ -4,31 +4,36 @@
  * SPDX-License-Identifier: MIT
  */
 
-#define DT_DRV_COMPAT zmk_behavior_macro
-
 #include <zephyr/device.h>
 #include <drivers/behavior.h>
 #include <zephyr/logging/log.h>
 #include <zmk/behavior.h>
 #include <zmk/behavior_queue.h>
 #include <zmk/keymap.h>
+#include <dt-bindings/zmk/macro.h>
 
 LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 
-#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+#if DT_HAS_COMPAT_STATUS_OKAY(zmk_behavior_macro) ||                                               \
```

Comment:
Could this be simplified using Kconfig + CMake, or would you rather do that in a separate commit?

Reply:
Yes, it can be! Rebased onto `main` where we have the `Kconfig.behaviors` and tweaked this to use the DT_HAS defines and simplify the code by only compiling if we have any of the macro DT compatibles enabled.